### PR TITLE
[Feature] Support Omni-source training on ImageNet and Kinetics dataset.

### PR DESCRIPTION
**Pull request by hukkai**

### CODE REVIEW

File: configs/recognition/omnisource/kinietics_imagenet.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+# dataset settings
+image_root = 'data/imagenet/'
+image_ann_train = 'meta/train.txt'
+
+video_root = 'data/kinetics400/videos_train'
+video_root_val = 'data/kinetics400/videos_val'
+video_ann_train = 'data/kinetics400/kinetics400_train_list_videos.txt'
+video_ann_val = 'data/kinetics400/kinetics400_val_list_videos.txt'
+
+num_images = 1281167  # number of training samples in ImageNet dataset
```
Comment (Dai-Wenxun) [inline configs/recognition/omnisource/kinietics_imagenet.py:35]:
```suggestion
num_images = 1281167  # number of training samples in the ImageNet dataset
```

File: configs/recognition/omnisource/kinietics_imagenet.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+batchsize_image = num_images // (num_iter * num_gpus)
+
+file_client_args = dict(
+    io_backend='petrel',
+    path_mapping=dict({
+        'data/kinetics400':
+        's3://openmmlab/datasets/action/Kinetics400',
+        'data/imagenet':
+        's3://openmmlab/datasets/classification/imagenet'
+    }))
```
Comment (Dai-Wenxun) [inline configs/recognition/omnisource/kinietics_imagenet.py:49]: Remove this args.

File: configs/recognition/omnisource/kinietics_imagenet.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        test_mode=True))
+
+file_client_args = dict(
+    backend='petrel',
+    path_mapping=dict({
+        'data/kinetics400':
+        's3://openmmlab/datasets/action/Kinetics400',
+        'data/imagenet':
+        's3://openmmlab/datasets/classification/imagenet'
+    }))
```
Comment (Dai-Wenxun) [inline configs/recognition/omnisource/kinietics_imagenet.py:135]: Remove this args.

File: mmaction/models/recognizers/recognizer_omni.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+import torch
+from mmengine.model import BaseModel
+
+from mmaction.registry import MODELS
+from mmaction.utils import ConfigType, ForwardResults, SampleList
+
+
+@MODELS.register_module()
+class RecognizerOmni(BaseModel):
+    """"""
```
Comment (cir7) [inline mmaction/models/recognizers/recognizer_omni.py:13]: To make it easier for users to understand, please add docstring to describe the main difference between RecognizerOmni and BaseRecognizer.

**Review by cir7 — APPROVED** [review]

**Pull request merged**